### ISSUE TITLE
[Enhancement] Code optimization.[HTTPClientHandler]

### ISSUE THREAD

**Issue by Alonexc** [apache/eventmesh#3958]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/eventmesh/issues?q=is%3Aissue) and found no similar issues.

### Enhancement Request

![image](https://github.com/apache/eventmesh/assets/91315508/1fc4c898-5883-4170-832b-3a7263648d14)
![image](https://github.com/apache/eventmesh/assets/91315508/fe1d0d20-f755-4c9d-a417-639219563bbb)
![image](https://github.com/apache/eventmesh/assets/91315508/93fa890a-ee50-49a5-bce6-6a53e35d5002)
![image](https://github.com/apache/eventmesh/assets/91315508/a8465397-3911-4021-baf4-0f4dfab173f0)
![image](https://github.com/apache/eventmesh/assets/91315508/f0c1350c-c13f-41ea-a4d6-2f6ff9c191e2)
![image](https://github.com/apache/eventmesh/assets/91315508/153417bc-328b-42ba-aeb7-67a4393703f8)
located at:
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/admin/handler/HTTPClientHandler.java line 65-68,81,87,108-109,143,154,169-175,118-126
analysis and explanation:
a. Method invocation may produce 'NullPointerException'
b. Use constant substitution in the `common` module constants class.
eventmesh-common/src/main/java/org/apache/eventmesh/common/enums/HttpMethod.java
eventmesh-runtime/src/main/java/org/apache/eventmesh/runtime/constants/EventMeshConstants.java
c. Excessive lambda usage

### Describe the solution you'd like

a. Use `Objects.requireNonNull()` to avoid NPE.
b. Constants are extracted into this class.[EventMeshConstants] (If someone has already done it, ignore this)
![image](https://github.com/apache/eventmesh/assets/91315508/da0468f0-2c35-41b8-a1b8-a4ebd3db1b73)

Replace with constants in the constants class.
![image](https://github.com/apache/eventmesh/assets/91315508/3e5d7f5a-3af3-441f-82c1-f78ba3769061)
![image](https://github.com/apache/eventmesh/assets/91315508/26f98a4c-7c8f-474f-88d1-eab4b0fcf4ac)
![image](https://github.com/apache/eventmesh/assets/91315508/3735498e-1cbc-4568-a9dc-e78448798d00)

we can also take `switch` and use the `Java Enum valueOf()` method.
e.g.
```
switch (HttpMethod.valueOf(httpExchange.getRequestMethod())) {
            case OPTIONS:
                preflight(httpExchange);
                break;
            case POST:
                post(httpExchange);
                break;
            case GET:
                get(httpExchange);
                break;
            default: //do nothing
                break;
        }
```

c. use `orElse()` method without lambda
![image](https://github.com/apache/eventmesh/assets/91315508/06bbba87-b7ce-4d9f-9166-59c8659334b2)

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

**Comment by nitheesh-daram** [comment]
hey @Alonexc i would like to work on this!

**Comment by nitheesh-daram** [comment]
Looks like #3955 , #3956 , #3957 , #3958 , #3959 , #3960 , #3961 , #3962  are similar changes , I'll work on these and raise a single PR instead of multiple PR's

**Comment by Alonexc** [comment]
@nitheesh-daram One pr corresponds to one issue, which helps managers to review.

**Comment by Alonexc** [comment]
> hey @Alonexc i would like to work on this!

Very welcome.

**Comment by nitheesh-daram** [comment]
> @nitheesh-daram One pr corresponds to one issue, which helps managers to review.

ok, thanks for the info.I'll raise the initial PR for #3955 with common strings as constants in EventmeshConstants.java file .once that is merged, I'll rebase my branches and I'll do the same for others.

is that ok?

**Comment by nitheesh-daram** [comment]
> Looks like #3955 , #3956 , #3957 , #3958 , #3959 , #3960 , #3961 , #3962 are similar changes , I'll work on these and raise a single PR instead of multiple PR's

Raised Separate PR for each issue. Due to common Dependencies in ```EventMeshConstants.java``` file, Conflicts will occur . I'll update/rebase the PR's Once at least one PR is merged.
